### PR TITLE
Add survival/nopickup.sc

### PR DESCRIPTION
**Pull request by KonaeAkira**
App that allows players to blacklist items that they do not want to pick up from the ground. Prevents unwanted items like rotten flesh from taking up precious inventory space.

### CODE REVIEW

File: programs/survival/nopickup.sc [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		print('There are no forbidden items');
+	);
+	return(null);
+);
+
+add(item) -> (
+	if (has(global_forbidden, str(item:0)),
+		print('[' + item:0 + '] is already forbidden');
+	,
+		put(global_forbidden, str(item:0), 0);
```
Comment (Ghoulboy78) [inline programs/survival/nopickup.sc:57]: why item:0? You can't input lists yet, only strings.

Comment (Ghoulboy78) [reply programs/survival/nopickup.sc:57]: I would suggest removing the `:0` in line 54 and 57 cos it's not a list here, it's a string.

Comment (KonaeAkira) [reply programs/survival/nopickup.sc:57]: With the new command types that scarpet supports, the input is an item (3-tuple of [name, count, nbt_data])

File: programs/survival/nopickup.sc [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+// returns how many of "item" is in the player's inventory
+_count_in_inventory(item) -> (
+	num_slots = inventory_size(player());
+	slot = -1;
+	count = 0;
+	while ((slot = inventory_find(player(), item:0, slot + 1)) != null, num_slots,
+		count += inventory_get(player(), slot):1;
+	);
+	return(count);
```
Comment (gnembon) [inline programs/survival/nopickup.sc:61]: don't need a 'return' statement at the end, just leave `count;` on the table. It also improves performance - returns are costly

File: programs/survival/nopickup.sc [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	if (global_uuid == null, _init()); // needed in case script is reloaded when players are connected
+	if (global_forbidden != {},
+		print('List of forbidden items:');
+		for (pairs(global_forbidden),
+			print('	[' + _:0 + ', ' + _:1 + ']');
+		);
+	,
+		print('There are no forbidden items');
+	);
+	return(null);
```
Comment (gnembon) [inline programs/survival/nopickup.sc:82]: ditto with return, but this is just a command so its executed only once per command, so not a big deal.

Comment (KonaeAkira) [reply programs/survival/nopickup.sc]: Thank you for your input! I've made a new commit implementing the changes you suggested.

**Review by gnembon — APPROVED** [review]

**Pull request merged**